launchdarkly/role_attributes_helper.go

```diff
@@ -0,0 +1,107 @@
+package launchdarkly
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ldapi "github.com/launchdarkly/api-client-go/v17"
+)
+
+func roleAttributesSchema(isDataSource bool) *schema.Schema {
+	return &schema.Schema{
+		Type: schema.TypeSet,
```

Comment:
I think we've run into issues with the TypeSet in the past. Do you remember that too? IIRC we went with TypeList for these kind of things instead.
```suggestion
		Type: schema.TypeList,
```

Reply:
hmm i don't fully remember, no. i had issues with the type list because it's a map on the LD end so the order was irregular, which caused unexpected diffs. i could pull something more complicated like we have to enforce ordering of nested environments on projects, but this seemed easier and that also seems to periodically fail. will ping you on slack now to discuss pros/cons